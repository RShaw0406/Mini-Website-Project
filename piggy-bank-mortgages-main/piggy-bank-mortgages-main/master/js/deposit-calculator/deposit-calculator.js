// -------------------- Calculate Deposit --------------------
let depositAmountRangeInput = document.getElementById('depositAmountRangeInput');
let depositAmountValue = document.getElementById('depositAmountValue');
let depositTimeRangeInput = document.getElementById('depositTimeRangeInput');
let depositTimeValue = document.getElementById('depositTimeValue');

export const calculateDeposit = (depositAmountRangeInput, depositTimeRangeInput) => {
  // Calculates the amount of time for a user to save up for a deposit
  let monthsInYear = 12;
  let totalMonthsForDeposit = monthsInYear * depositTimeRangeInput.value;
  let totalMonthlySaving = depositAmountRangeInput.value / totalMonthsForDeposit;
  let monthlySavingsValue = document.getElementById('monthlySavingsValue');
  monthlySavingsValue.innerHTML = '£' + totalMonthlySaving.toFixed(2);
};

calculateDeposit(depositAmountRangeInput, depositTimeRangeInput);

depositAmountValue.innerHTML = '£' + depositAmountRangeInput.value;
depositAmountRangeInput.oninput = () => {
  depositAmountValue.innerHTML = '£' + depositAmountRangeInput.value;
  calculateDeposit(depositAmountRangeInput, depositTimeRangeInput);
};

depositTimeValue.innerHTML = depositTimeRangeInput.value + ' years';
depositTimeRangeInput.oninput = () => {
  depositTimeValue.innerHTML = depositTimeRangeInput.value + ' years';
  calculateDeposit(depositAmountRangeInput, depositTimeRangeInput);
};

// -------------------- Control Deposit Info Modals --------------------
let modalDepositInfo = document.getElementById('modalDepositInfo');
let btnOpenModalDepositAmountInfo = document.getElementById('btnOpenModalDepositAmountInfo');
let btnOpenModalDepositTimeInfo = document.getElementById('btnOpenModalDepositTimeInfo');
let finishedDepositInfo = document.getElementById('finishedDepositInfo');
let headerDepositAmountInfo = document.getElementById('headerDepositAmountInfo');
let headerDepositTimeInfo = document.getElementById('headerDepositTimeInfo');
let divDepositAmountInfo = document.getElementById('divDepositAmountInfo');
let divDepositTimeInfo = document.getElementById('divDepositTimeInfo');

btnOpenModalDepositAmountInfo.onclick = () => {
  modalDepositInfo.style.display = 'block';
  headerDepositAmountInfo.style.display = 'block';
  headerDepositTimeInfo.style.display = 'none';
  divDepositAmountInfo.style.display = 'block';
  divDepositTimeInfo.style.display = 'none';
};

btnOpenModalDepositTimeInfo.onclick = () => {
  modalDepositInfo.style.display = 'block';
  headerDepositAmountInfo.style.display = 'none';
  headerDepositTimeInfo.style.display = 'block';
  divDepositAmountInfo.style.display = 'none';
  divDepositTimeInfo.style.display = 'block';
};

finishedDepositInfo.onclick = () => {
  modalDepositInfo.style.display = 'none';
};

window.onclick = (event) => {
  if (event.target === modalDepositInfo) {
    modalDepositInfo.style.display = 'none';
  }
};
