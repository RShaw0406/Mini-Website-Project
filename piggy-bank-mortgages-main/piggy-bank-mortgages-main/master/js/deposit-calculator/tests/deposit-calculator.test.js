import * as fs from 'fs';

document.body.innerHTML = fs.readFileSync(`${__dirname}/resources/test.html`, 'utf-8');
const btnOpenModalDepositAmountInfo = document.getElementById('btnOpenModalDepositAmountInfo');
const modalDepositInfo = document.getElementById('modalDepositInfo');
const headerDepositAmountInfo = document.getElementById('headerDepositAmountInfo');
const headerDepositTimeInfo = document.getElementById('headerDepositTimeInfo');
const divDepositAmountInfo = document.getElementById('divDepositAmountInfo');
const divDepositTimeInfo = document.getElementById('divDepositTimeInfo');
const closeModalDepositInfo = document.getElementsByClassName('close')[0];
const finishedDepositInfo = document.getElementById('finishedDepositInfo');



describe('Deposit Calculator tests', () => {
  describe('When calculateDeposit is called', () => {
    it('should update the html correctly', () => {
      // Arrange
      let depositAmountRangeInput = document.getElementById('depositAmountRangeInput');
      depositAmountRangeInput.value = 50000
      let depositTimeRangeInput = document.getElementById('depositTimeRangeInput');
      depositTimeRangeInput.value = 20
      const depositCalculator = require('../deposit-calculator.js');

      // Act
      depositCalculator.calculateDeposit(depositAmountRangeInput, depositTimeRangeInput)
      // Assert
      expect(document.body.innerHTML).toMatchSnapshot()
    });
  });
  describe('When btnOpenModalDepositAmountInfo is clicked', () => {
    it('should update the element styles correctly', () => {
      // Arrange
      require('../deposit-calculator.js');
      // Act
      btnOpenModalDepositAmountInfo.click();
      // Assert
      expect(modalDepositInfo.style.display).toEqual('block')
      expect(headerDepositAmountInfo.style.display).toEqual('block')
      expect(headerDepositTimeInfo.style.display).toEqual('none')
      expect(divDepositAmountInfo.style.display).toEqual('block')
      expect(divDepositTimeInfo.style.display).toEqual('none')
    });
  });
  describe('When btnOpenModalDepositTimeInfo is clicked', () => {
    it('should update the element styles correctly', () => {
      // Arrange
      require('../deposit-calculator.js');
      // Act
      btnOpenModalDepositTimeInfo.click();
      // Assert
      expect(modalDepositInfo.style.display).toEqual('block')
      expect(headerDepositAmountInfo.style.display).toEqual('none')
      expect(headerDepositTimeInfo.style.display).toEqual('block')
      expect(divDepositAmountInfo.style.display).toEqual('none')
      expect(divDepositTimeInfo.style.display).toEqual('block')
    });
  });
  describe('When finishedDepositInfo is clicked', () => {
    it('should update the element styles correctly', () => {
      // Arrange
      require('../deposit-calculator.js');
      // Act
      finishedDepositInfo.click();
      // Assert
      expect(modalDepositInfo.style.display).toEqual('none')
    });
  });
  describe('When a value is input into depositAmountRangeInput', () => {
    it('should update the html correctly', () => {
      // Arrange
      const depositAmountRangeInput = document.getElementById('depositAmountRangeInput');
      require('../deposit-calculator.js');
      const event = {target: {value: "test"}}
      // Act
      depositAmountRangeInput.oninput(event)
      // Assert
      expect(document.body.innerHTML).toMatchSnapshot()
    });
  });
  describe('When a value is input into depositTimeRangeInput', () => {
    it('should update the html correctly', () => {
      // Arrange
      const depositTimeRangeInput = document.getElementById('depositTimeRangeInput');
      require('../deposit-calculator.js');
      const event = {target: {value: "test"}}
      // Act
      depositTimeRangeInput.oninput(event)
      // Assert
      expect(document.body.innerHTML).toMatchSnapshot()
    });
  });
  describe('When the window is clicked', () => {
    it('should update the element styles correclty if event target is modalDepositInfo', () => {
      // Arrange
      const event = {target: modalDepositInfo}
      require('../deposit-calculator.js');
      // Act
      document.defaultView.onclick(event)
      // Assert
      expect(modalDepositInfo.style.display).toEqual('none')
    });
  });
});
