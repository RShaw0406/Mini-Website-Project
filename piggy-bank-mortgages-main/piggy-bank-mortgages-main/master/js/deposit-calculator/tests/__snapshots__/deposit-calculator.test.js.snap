// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deposit Calculator tests When a value is input into depositAmountRangeInput should update the html correctly 1`] = `
"<label for="depositAmountRangeInput"></label><input id="depositAmountRangeInput" type="range" value="1000">
<div id="depositAmountValue">£100</div>
<label for="depositTimeRangeInput"></label><input id="depositTimeRangeInput" type="number" value="20">
<div id="depositTimeValue">20 years</div>
<div id="monthlySavingsValue">£0.42</div>
<div id="modalDepositInfo" style="display: none;"></div>
<button id="btnOpenModalDepositAmountInfo"></button>
<button id="btnOpenModalDepositTimeInfo"></button>
<button class="close"></button>
<button id="finishedDepositInfo"></button>
<div id="headerDepositAmountInfo" style="display: none;"></div>
<div id="headerDepositTimeInfo" style="display: block;"></div>
<div id="divDepositAmountInfo" style="display: none;"></div>
<div id="divDepositTimeInfo" style="display: block;"></div>
"
`;

exports[`Deposit Calculator tests When a value is input into depositTimeRangeInput should update the html correctly 1`] = `
"<label for="depositAmountRangeInput"></label><input id="depositAmountRangeInput" type="range" value="1000">
<div id="depositAmountValue">£100</div>
<label for="depositTimeRangeInput"></label><input id="depositTimeRangeInput" type="number" value="20">
<div id="depositTimeValue">20 years</div>
<div id="monthlySavingsValue">£0.42</div>
<div id="modalDepositInfo" style="display: none;"></div>
<button id="btnOpenModalDepositAmountInfo"></button>
<button id="btnOpenModalDepositTimeInfo"></button>
<button class="close"></button>
<button id="finishedDepositInfo"></button>
<div id="headerDepositAmountInfo" style="display: none;"></div>
<div id="headerDepositTimeInfo" style="display: block;"></div>
<div id="divDepositAmountInfo" style="display: none;"></div>
<div id="divDepositTimeInfo" style="display: block;"></div>
"
`;

exports[`Deposit Calculator tests When calculateDeposit is called should update the html correctly 1`] = `
"<label for="depositAmountRangeInput"></label><input id="depositAmountRangeInput" type="range" value="1000">
<div id="depositAmountValue">£100</div>
<label for="depositTimeRangeInput"></label><input id="depositTimeRangeInput" type="number" value="20">
<div id="depositTimeValue">20 years</div>
<div id="monthlySavingsValue">£0.42</div>
<div id="modalDepositInfo"></div>
<button id="btnOpenModalDepositAmountInfo"></button>
<button id="btnOpenModalDepositTimeInfo"></button>
<button class="close"></button>
<button id="finishedDepositInfo"></button>
<div id="headerDepositAmountInfo"></div>
<div id="headerDepositTimeInfo"></div>
<div id="divDepositAmountInfo"></div>
<div id="divDepositTimeInfo"></div>
"
`;
